pkg/watermark/publish/publisher.go

```diff
@@ -242,7 +248,13 @@ func (p *publish) loadLatestFromStore() wmb.Watermark {
 
 // GetLatestWatermark returns the latest watermark for that processor.
 func (p *publish) GetLatestWatermark() wmb.Watermark {
-	return p.GetHeadWM()
+	var latestWatermark = wmb.Watermark(time.UnixMilli(-1))
+	for _, wm := range p.headWatermarks {
+		if wm.After(time.Time(latestWatermark)) {
+			latestWatermark = wm
+		}
```

Comment:
How to compare the watermark will be updated later when we finalize the design?

Reply:
this method is only used in tests.